mqtt/mqtt-edgehub/src/auth/authorization/edgehub.rs

```diff
@@ -14,23 +14,32 @@ use mqtt_broker::{
 /// telemetry messages, etc...
 ///
 /// For non-iothub-specific primitives it delegates the request to an inner authorizer (`PolicyAuthorizer`).
-#[derive(Debug)]
 pub struct EdgeHubAuthorizer<Z> {
     iothub_allowed_topics: RefCell<HashMap<ClientId, Vec<String>>>,
     identities_cache: HashMap<ClientId, IdentityUpdate>,
     inner: Z,
+    broker_ready: Option<BrokerReadyHandle>,
 }
 
 impl<Z, E> EdgeHubAuthorizer<Z>
 where
     Z: Authorizer<Error = E>,
     E: StdError,
 {
-    pub fn new(authorizer: Z) -> Self {
+    pub fn new(authorizer: Z, broker_ready: BrokerReadyHandle) -> Self {
+        Self::create(authorizer, Some(broker_ready))
+    }
+
+    pub fn without_ready_handle(authorizer: Z) -> Self {
```

Comment:
nit: vice versa? 
```
new(inner: Z) -> Self
with_ready_handle(inner: Z, ready_handle: BrokerReadyHandle) -> Self
```

Reply:
it is supposed to be used with ready_handle always. however, for tests, it adds some inconvenience.